src/main/scala/rocket/CSR.scala

```diff
@@ -851,6 +856,7 @@ class CSRFile(
     when (reset) {
       bpc.action := false
       bpc.dmode := false
+      bpc.chain := false
```

Comment:
is this the bit that prevents longer chains? Is the idea that only even/odd are allowed to have the chain bit set?

Reply:
This is just the bit indicating the breakpoint is chained.  I didn't restrict it to even/odd pairs; chains can still begin on odd numbers but are always limited to 2.  The reason I'm resetting the bits is to prevent them from coming up in an inconsistent state where >2 breakpoints appear to be chained.

The hardware is a little simpler if it's restricted to even-odd pairs, so if you think that won't cause problems, I could revise this PR to incorporate that restriction.  I think the only case where the even-odd restriction actually reduces the usefulness of the breakpoints is when there is an odd number of breakpoints, and D-mode wants to chain two of them but share some with M-mode.

Reply:
Nah, I don't think you need to do the even/odd thing. So how does it work to limit the chain length? Does it prohibit you from setting the bit if the previous one is set, or something?

Reply:
Right, the line

    bp.control.chain := newBPC.chain && !(prevChain || nextChain) && (dMode || !nextDMode)

prevents setting the chain bit when either the previous or next breakpoint has its chain bit set.

Reply:
got it